src/features/ProductInfoPage/components/ProductInfoPage.jsx

```diff
@@ -0,0 +1,105 @@
+import React, {useState, useEffect} from 'react';
+import Footer from '../../Footer/components/Footer';
+import {fetchProductById} from '../../../common/utils/api';
+import {useParams} from 'react-router';
+import {AiOutlineHeart} from 'react-icons/ai';
+import {BiLinkExternal} from 'react-icons/bi';
+import {addToCart} from '../../CartPage/slices/cartSlice';
+import {useDispatch} from 'react-redux';
+import {setSnackbarConfig} from '../../../common/slices/snackbarSlice';
+
+const ProductInfoPage = () => {
+  const [product, setProduct] = useState(null);
+  const [quantity, setQuantity] = useState(1);
+  const {id} = useParams();
+  const dispatch = useDispatch();
+
+  useEffect(
+      () => {
+        setProductById();
+      },
+      [],
+  );
+
+  const setProductById = async () => {
```

Comment:
move to thunk file, no api calls in jsx files. 

please remember the goal of the assignment is not to check whether you can develop the app, it is understood the candidate is well-skilled and experienced before the allocation of assignment. 

the goal of the assignment is to check whether the candidate can follow the rules and conventions set in the project obediently.

Reply:
Not following thunks structure in this assignment.

Reply:
It appears that the assignment mentions the use of redux-toolkit, which comes with thunk support by default. Whether or not to use the thunk creators by toolkit or to use you own thunk functions are the choice of the candidate, but using thunks via redux toolkit appears mandatory.

Please use thunks in all API calls, because we will be doing it all the times in our prod application, moreover, you have been given an overview of thunks in mcp project, you can reference the examples from here:

https://github.com/browserstack/frontend/blob/04bd0b849492b7cfac75d9823bf9f14693190231/apps/mcspt/src/features/Home/slices/newPerformanceSessionThunks.js

Reply:
I understand. By making code changes for thunks here may increase scope, which we don't have right now. I will remember this for FE tasks.